x-pack/legacy/plugins/infra/public/containers/metrics_explorer/use_metrics_explorer_data.ts

```diff
@@ -96,6 +96,9 @@ export function useMetricsExplorerData(
       }
       setLoading(false);
     })();
+
+    // TODO: fix this dependency list while preserving the semantics
```

Comment:
@simianhacker @phillipb I couldn't reliably decompose this effect to derive a correct dependency list. I'd appreciate your help here.

Reply:
@weltenwort Yeah, this is pretty gnarly. Does `useEffect` really need to have all of it's deps specified? I can think of use cases where you might want to use some data in an effect, but you don't necessarily want to rerun that effect every time the data changes.

Reply:
Assuming for now that it does, I'll need to spend a bit of time understanding this more and refactoring it. Are you ok, with me taking that on as a separate PR/issue?

Reply:
Absolutely, let's take it to a different PR. I was planning to share some strategies for dealing with it in a session some time soon. In the meantime https://overreacted.io/a-complete-guide-to-useeffect/ is a nice deep dive into the topic.